hermes/probe/create_file.go

```diff
@@ -6,109 +6,119 @@ import (
+	if r.readDone() {
 		return 0, io.EOF
 	}
-	if len(b) > r.size-r.i {
-		n, err = r.rand.Read(b[:r.size-r.i])
-	} else {
-		n, err = r.rand.Read(b)
+	b := buff
+	if r.bufferTooLong(b){
+		b = buff[:r.bytesLeft()]
+	}
+	n, err = r.rand.Read(b)
+	if(err!=nil){
+		return 0, err // 0 as nil casn't be returned as a type int argument
 	}
 	r.i += n
 	return
 }
 
-func (f *RandomFile) generateChecksum() (string, error) {
+// Warning: NewReader is not thread safe
+func (f *RandomHermesFile) NewReader() *randomHermesFileReader {
+	return &randomHermesFileReader{size: f.Size, rand: rand.New(rand.NewSource(f.ID))} // ID will serve as a Seed and i will be set to 0 automatically
+}
+
+func (f *RandomHermesFile) CheckSum() ([]byte, error) {
 	r := f.NewReader()
 	h := sha1.New()
-	buff := make([]byte, 8)
-	for {
-		n, err := r.Read(buff)
-		if err == io.EOF {
-			break
-		}
-		if _, err = io.Copy(h, r); err != nil {
-			return nil, fmt.Errorf("io.Copy: %v", err)
-		}
-
+	if _, err := io.Copy(h, r); err != nil {
+		return nil, fmt.Errorf("io.Copy: %v", err)
 	}
-
 	// returns checksum in hex notation
-	return fmt.Sprintf("%x", h.Sum(nil)), nil
+	return h.Sum(nil), nil
 }
 
-func (f *RandomFile) generateFileName(fileID int) (string, error) {
-	checksum, err := f.generateChecksum()
+func (f *RandomHermesFile) FileName() (string, error) {
+	if f.ID < begin || f.ID > end {
+		return "", fmt.Errorf("The file ID provided %v wasn't in the required range [1,50]", f.ID)
+	}
+	if f.Size > maxFileSize {
+		return "", fmt.Errorf("The file size provided %v bytes exceeded the limit %v bytes", f.Size, maxFileSize)
+	}
+	if f.Size <= 0 {
+		return "", fmt.Errorf("The file size provided %v is not a positive number as required", f.Size)
+	}
+	checksum, err := f.CheckSum()
 	if err != nil {
-		return nil, err
+		return "", err
 	}
-	return fmt.Sprintf("Hermes_%02d_%v", fileID, checksum), nil
+	return fmt.Sprintf("Hermes_%02d_%v", f.ID, fmt.Sprintf("%x", checksum)), nil
 }
 
-func (p *MonitorProbe) CreateFile(ctx context.Context, target string, bucketName string, fileID int, fileSize int) error {
-	if fileID < 1 || fileID > 50 {
-		return fmt.Errorf("At %v the file ID provided wasn't in the required range [0,50]", time.Now())
-	}
-	f := RandomFile{Seed: fileID, Size: fileSize}
-	fileName, err := f.generateFileName(fileID)
+func (p *MonitorProbe) CreateFile(ctx context.Context, target string, bucketName string, fileID int64, fileSize int, client stiface.Client) error {
+	f := RandomHermesFile{ID: fileID, Size: fileSize}
+	fileName, err := f.FileName()
+	fmt.Println("Got Here File successfully named")
 	if err != nil {
 		return err
 	}
-	client, err := storage.NewClient(ctx)
+	/*client, err := storage.NewClient(ctx)
 	if err != nil {
 		return fmt.Errorf("storage.NewClient: %v", err)
 	}
-	defer client.Close()
-	ctx, cancel := context.WithTimeout(ctx, p.opts.Timeout)
-	defer cancel()
+	defer client.Close()*/
+	//ctx, cancel := context.WithTimeout(ctx, p.opts.Timeout)
```

Comment:
Remove this, rather than commenting it out (here & elsewhere).

Reply:
noted! git replaced with and committed an old version not ready for a review